hooks/test/browser/useState.test.js

```diff
@@ -211,4 +211,24 @@ describe('useState', () => {
 		rerender();
 		expect(scratch.innerHTML).to.equal('');
 	});
+
+	it('should render a second time when the render function updates state', () => {
+		const calls = [];
+		const App = () => {
+			const [greeting, setGreeting] = useState('bye');
+
+			if (greeting === 'bye') {
+				setGreeting('hi');
+			}
+
+			calls.push(greeting);
+
+			return <p>{greeting}</p>;
+		};
+
+		render(<App />, scratch);
```

Comment:
To ensure no additional enqueued component invocations:

```suggestion
		act(() => {
			render(<App />, scratch);
		});
```